flight-search-be/src/main/java/com/encora/flight_search_be/client/AmadeusClient.java

```diff
@@ -0,0 +1,98 @@
+package com.encora.flight_search_be.client;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Value; 
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class AmadeusClient {
+    
+    private final RestTemplate restTemplate = new RestTemplate();
+    private String accessToken;
+    private long tokenExpiration;
+
+    @Value("${amadeus.api.client-id}")
+    private String clientId;
+
+    @Value("${amadeus.api.client-secret}")
+    private String clientSecret;
+
+    private final String tokenUrl = "https://test.api.amadeus.com/v1/security/oauth2/token";
+    private final String flightSearchUrl = "https://test.api.amadeus.com/v2/shopping/flight-offers";
+
+    public String getAccessToken() {
+        if (accessToken == null || System.currentTimeMillis() > tokenExpiration) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+            String body = "grant_type=client_credentials" +
```

Comment:
Suggestion: Try to have the strings on a constant variable, you could have it as a template and then fill the needed variables.